includes/REST_API/Stories_Users_Controller.php

```diff
@@ -94,6 +94,28 @@ public static function get_registration_action_priority(): int {
 		return 100;
 	}
 
+	/**
+	 * Permissions check for getting all users.
+	 *
+	 * Allows edit_posts capabilities queries for stories if the user has the same cap,
+	 * enabling them to see the users dropdown.
+	 *
+	 * @since 1.28.1
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return true|WP_Error True if the request has read access, otherwise WP_Error object.
+	 */
+	public function get_items_permissions_check( $request ) {
+		if (
+			! empty( $request['capabilities'] ) &&
+			[ $this->story_post_type->get_cap_name( 'edit_posts' ) ] === $request['capabilities']
+		) {
+			unset( $request['capabilities'] );
+		}
```

Comment:
```suggestion
		/**
		 * Capabilities.
		 *
		 * @var string[]|null $capabilities
		 */
		$capabilities = $request['capabilities'];
		if ( ! empty( $capabilities ) ) {
			$request['capabilities'] = array_filter(
				$capabilities,
				fn( $cap ) => $cap !== $this->story_post_type->get_cap_name( 'edit_posts' )
			);
		}
```

This might work better, as it would remove web stories capability check from any array.